src/PowerShellEditorServices/Services/DebugAdapter/Handlers/SemanticTokens.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.PowerShell.EditorServices.Services;
+using Microsoft.PowerShell.EditorServices.Services.TextDocument;
+using System.Management.Automation.Language;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document.Proposals;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models.Proposals;
+
+namespace Microsoft.PowerShell.EditorServices.Handlers
+{
+#pragma warning disable 618
+    internal class SemanticTokens : SemanticTokensHandler
+    {
+        private readonly ILogger _logger;
+        public WorkspaceService _workspaceService;
+        public SemanticTokens(ILogger<SemanticTokens> logger, WorkspaceService workspaceService) : base(new SemanticTokensRegistrationOptions() {
+            DocumentSelector = DocumentSelector.ForLanguage("powershell"),
+            Legend = new SemanticTokensLegend(),
+            DocumentProvider = new Supports<SemanticTokensDocumentProviderOptions>(true,
+                new SemanticTokensDocumentProviderOptions() {
+                    Edits = true
+                }),
+            RangeProvider = true
+        })
+        {
+            _logger = logger;
+            _workspaceService = workspaceService;
+        }
+
+        public override async Task<OmniSharp.Extensions.LanguageServer.Protocol.Models.Proposals.SemanticTokens> Handle(
+            SemanticTokensParams request, CancellationToken cancellationToken)
+        {
+            var result = await base.Handle(request, cancellationToken);
+            return result;
+        }
+
+        public override async Task<SemanticTokensOrSemanticTokensEdits> Handle(SemanticTokensEditsParams request,
+            CancellationToken cancellationToken)
+        {
+            var result = await base.Handle(request, cancellationToken);
+            return result;
+        }
+
+        protected override async Task Tokenize(SemanticTokensBuilder builder, ITextDocumentIdentifierParams identifier,
+            CancellationToken cancellationToken)
+        {
+            ScriptFile file = _workspaceService.GetFile(DocumentUri.GetFileSystemPath(identifier));
+            await Task.Yield();
+            Token[] tokens = file.ScriptTokens;
+            foreach (var token in tokens){
+                pushToken(token, builder);
+            }
+        }
```

Comment:
An alternate implementation might be:

```suggestion
        protected override Task Tokenize(SemanticTokensBuilder builder, ITextDocumentIdentifierParams identifier,
            CancellationToken cancellationToken)
        {
            ScriptFile file = _workspaceService.GetFile(DocumentUri.GetFileSystemPath(identifier));
            Token[] tokens = file.ScriptTokens;
            foreach (var token in tokens){
                pushToken(token, builder);
            }
            return Task.CompletedTask;
        }
```

Reply:
Context: This is how you can return a Task without needing an async/await in your method.